### PR TITLE
Add support for setting autoupdate in quadlet

### PR DESCRIPTION
**Pull request by rhatdan**
#### Does this PR introduce a user-facing change?

```release-note
Quadlet .container format now supports AutoUpdate option.
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 1]
```diff
@@ -170,6 +171,14 @@ similar to `Environment`.
 
 This key can be listed multiple times.
 
+### `AutoUpdate=`
+
+Indicates whether the container will be autoupdated.  The following values are supported:
```
Comment (vrothberg) [inline docs/source/markdown/podman-systemd.unit.5.md:176]: Can you point to `podman-auto-update(1)` for more details?

**Review by vrothberg — APPROVED** [review]
LGTM

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 2]
```diff
@@ -839,3 +848,4 @@ Label=org.test.Key=value
 **[systemd.service(5)](https://www.freedesktop.org/software/systemd/man/systemd.service.html)**,
 **[podman-run(1)](podman-run.1.md)**,
 **[podman-network-create(1)](podman-network-create.1.md)**
+**[podman-auto-update(1)] (podman-auto-update.1.md)**
```
Comment (Luap99) [inline docs/source/markdown/podman-systemd.unit.5.md:851]: need to remove the whitespace

File: docs/source/markdown/podman-systemd.unit.5.md [hunk 3]
```diff
@@ -170,6 +171,14 @@ similar to `Environment`.
 
 This key can be listed multiple times.
 
+### `AutoUpdate=`
+
+Indicates whether the container will be auto-updated (`[podman-auto-update(1)](podman-auto-update.1.md)`). The following values are supported:
```
Comment (Luap99) [inline docs/source/markdown/podman-systemd.unit.5.md:176]: also here, you need to remove the code block, the markdown link will not work inside it

**Review by Luap99 — COMMENTED** [review]
/lgtm

**Pull request merged**